### PR TITLE
Split PoET off into separate repo from Sawtooth Core

### PR DESCRIPTION
**Pull request by aludvik**
This commit sets up the Sawtooth PoET repo by reorganizing the Sawtooth Core repo post `git filter-branch --subdirectory-filter consensus/poet`.

### CODE REVIEW

File: bin/generate_cli_output [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+save_usage sawtooth-rest-api
+save_usage settings-tp
+save_usage identity-tp
+
+save_usage poet
+save_usage poet enclave
+save_usage poet registration
+save_usage poet registration create
+
+save_usage intkey
```
Comment (peterschwarz) [inline bin/generate_cli_output]: Should these be included?

Comment (aludvik) [reply bin/generate_cli_output]: They are removed in a later commit

**Comment by aludvik** [comment]
@dcmiddle Added a commit with BUILD.md and dropped the commands from `docker-compose.yaml`.

**Comment by dcmiddle** [comment]
Thanks for the quick patch @aludvik. Unfortunately got another odd build error from `docker-compose-installed.yaml`
```
Step 5/30 : RUN cd /project/common/  && /project/bin/protogen  && if [ -d "debian" ]; then rm -rf debian; fi  && python3 setup.py clean --all  && python3 setup.py --command-packages=stdeb.command debianize  && if [ -d "packaging/ubuntu" ]; then cp -R packaging/ubuntu/* debian/; fi  && dpkg-buildpackage -b -rfakeroot -us -uc
 ---> Running in 27e5471f070b
ERROR: VERSION file and (bumped?) git describe versions differ: 1.1.2 != 1.1.1
Traceback (most recent call last):
  File "setup.py", line 25, in <module>
    ['../bin/get_version']).decode('utf-8').strip(),
  File "/usr/lib/python3.5/subprocess.py", line 626, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.5/subprocess.py", line 708, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['../bin/get_version']' returned non-zero exit status 1
ERROR: Service 'poet-engine' failed to build: The command '/bin/sh -c cd /project/common/  && /project/bin/protogen  && if [ -d "debian" ]; then rm -rf debian; fi  && python3 setup.py clean --all  && python3 setup.py --command-packages=stdeb.command debianize  && if [ -d "packaging/ubuntu" ]; then cp -R packaging/ubuntu/* debian/; fi  && dpkg-buildpackage -b -rfakeroot -us -uc' returned a non-zero code: 1
```

**Pull request merged**